# Intro to Lists

Organize your data so you can work with it efficiently.

## Introduction

When doing data science, you need a way to organize your data so you can work with it efficiently. Python has many data structures available for holding your data, such as lists, sets, dictionaries, and tuples. In this tutorial, you will learn how to work with Python lists.

## Motivation

In the Petal to the Metal competition, your goal is to classify the species of a flower based only on its image. (This is a common task in computer vision, and it is called image classification.) Towards this goal, say you organize the names of the flower species in the data.

In [1]:
flowers = "pink primrose,hard-leaved pocket orchid,canterbury bells,sweet pea,english marigold,tiger lily,moon orchid,bird of paradise,monkshood,globe thistle"

print(type(flowers))
print(flowers)

<class 'str'>
pink primrose,hard-leaved pocket orchid,canterbury bells,sweet pea,english marigold,tiger lily,moon orchid,bird of paradise,monkshood,globe thistle


Kode ini akan menunjukkan jenis data dari variabel flowers, yang seharusnya menjadi str (string), dan kemudian mencetak isi dari variabel tersebut. Dalam hal ini, variabel flowers berisi daftar nama bunga yang dipisahkan oleh koma.

In [2]:
flowers_list = ["pink primrose", "hard-leaved pocket orchid", "canterbury bells", "sweet pea", "english marigold", "tiger lily", "moon orchid", "bird of paradise", "monkshood", "globe thistle"]

print(type(flowers_list))
print(flowers_list)

<class 'list'>
['pink primrose', 'hard-leaved pocket orchid', 'canterbury bells', 'sweet pea', 'english marigold', 'tiger lily', 'moon orchid', 'bird of paradise', 'monkshood', 'globe thistle']


Kode ini akan menunjukkan jenis data dari variabel `flowers_list`, yang seharusnya menjadi `list`, dan kemudian mencetak isi dari variabel tersebut. Dalam hal ini, variabel `flowers_list` berisi daftar nama bunga yang dipisahkan oleh koma.

## Lists

### Length

We can count the number of entries in any list with len(), which is short for "length". You need only supply the name of the list in the parentheses.

In [3]:
# The list has ten entries
print(len(flowers_list))

10


 print(len(flowers_list)) digunakan untuk mencetak panjang daftar, yang merupakan jumlah entri dalam daftar tersebut.

### Indexing

We can refer to any item in the list according to its position in the list (first, second, third, etc). This is called indexing.

In [4]:
print("First entry:", flowers_list[0])
print("Second entry:", flowers_list[1])

# The list has length ten, so we refer to final entry with 9
print("Last entry:", flowers_list[9])

First entry: pink primrose
Second entry: hard-leaved pocket orchid
Last entry: globe thistle


Kode ini akan mencetak entri pertama (flowers_list[0]), kedua (flowers_list[1]), dan terakhir (flowers_list[9]) dari daftar flowers_list. Komentar di baris ke-4 menjelaskan bahwa karena panjang daftar adalah sepuluh, dapat merujuk pada entri terakhir dengan indeks 9.

### Slicing

You can also pull a segment of a list (for instance, the first three entries or the last two entries). This is called slicing.

In [5]:
print("First three entries:", flowers_list[:3])
print("Final two entries:", flowers_list[-2:])

First three entries: ['pink primrose', 'hard-leaved pocket orchid', 'canterbury bells']
Final two entries: ['monkshood', 'globe thistle']


Kode ini menggunakan slicing untuk memilih tiga entri pertama (flowers_list[:3]) dan dua entri terakhir (flowers_list[-2:]) dari daftar flowers_list. Hasilnya akan dicetak ke layar.

### Removing items

Remove an item from a list with .remove(), and put the item you would like to remove in parentheses.

In [6]:
flowers_list.remove("globe thistle")
print(flowers_list)

['pink primrose', 'hard-leaved pocket orchid', 'canterbury bells', 'sweet pea', 'english marigold', 'tiger lily', 'moon orchid', 'bird of paradise', 'monkshood']


Kode ini akan menghapus entri "globe thistle" dari flowers_list menggunakan metode remove() dan kemudian mencetak daftar flowers_list yang telah diperbarui.

### Adding items

Add an item to a list with .append(), and put the item you would like to add in parentheses.

In [7]:
flowers_list.append("snapdragon")
print(flowers_list)

['pink primrose', 'hard-leaved pocket orchid', 'canterbury bells', 'sweet pea', 'english marigold', 'tiger lily', 'moon orchid', 'bird of paradise', 'monkshood', 'snapdragon']


Kode ini menggunakan metode append() untuk menambahkan entri baru "snapdragon" ke flowers_list, dan kemudian mencetak daftar flowers_list yang telah diperbarui.

### Lists are not just for strings

we have only worked with lists where each item in the list is a string. But lists can have items with any data type, including booleans, integers, and floats.

In [8]:
hardcover_sales = [139, 128, 172, 139, 191, 168, 170]

Dengan kode ini, list hardcover_sales akan berisi data penjualan buku keras untuk beberapa periode waktu tertentu.

In [9]:
print("Length of the list:", len(hardcover_sales))
print("Entry at index 2:", hardcover_sales[2])

Length of the list: 7
Entry at index 2: 172


Kode ini akan mencetak panjang list `hardcover_sales` dan nilai entri pada indeks ke-2. Misalnya, jika nilai pada indeks ke-2 adalah 172, maka keluaran akan menjadi:

Length of the list: 7
Entry at index 2: 172

In [10]:
print("Minimum:", min(hardcover_sales))
print("Maximum:", max(hardcover_sales))

Minimum: 128
Maximum: 191


Kode ini akan mencetak nilai minimum dan maksimum dari list `hardcover_sales`. Misalnya, jika list tersebut adalah `[139, 128, 172, 139, 191, 168, 170]`, maka keluarannya akan menjadi:

Minimum: 128
Maximum: 191

Ini menunjukkan nilai minimum penjualan buku keras (128) dan nilai maksimum penjualan buku keras (191) dalam periode waktu yang diberikan.

In [11]:
print("Total books sold in one week:", sum(hardcover_sales))

Total books sold in one week: 1107


Kode ini menggunakan fungsi `sum()` untuk menghitung jumlah total semua nilai dalam list `hardcover_sales`. Kemudian, hasilnya dicetak dengan pesan yang sesuai. Misalnya, jika list tersebut adalah `[139, 128, 172, 139, 191, 168, 170]`, maka keluarannya akan menjadi:

Total books sold in one week: 1107

Ini menunjukkan bahwa total 1107 buku terjual dalam satu minggu.

In [12]:
print("Average books sold in first five days:", sum(hardcover_sales[:5])/5)

Average books sold in first five days: 153.8


Kode ini menggunakan slicing untuk memilih nilai penjualan buku keras dalam lima hari pertama (`hardcover_sales[:5]`). Kemudian, nilai-nilai tersebut dijumlahkan dan dibagi dengan jumlah hari (5) untuk mendapatkan rata-rata. Hasilnya kemudian dicetak dengan pesan yang sesuai.

Misalnya, jika list tersebut adalah `[139, 128, 172, 139, 191, 168, 170]`, maka keluarannya akan menjadi:

Average books sold in first five days: 153.8

Ini menunjukkan bahwa rata-rata penjualan buku keras dalam lima hari pertama adalah 153.8 buku per hari.